src/Estimators/tests/test_local_energy_est.cpp

```diff
@@ -87,7 +84,7 @@ TEST_CASE("LocalEnergy with hdf5", "[estimators]")
   (*W.begin())->Properties(WP::LOCALENERGY)    = 1.1;
   (*W.begin())->Properties(WP::LOCALPOTENTIAL) = 1.2;
 
-  std::vector<observable_helper*> h5desc;
```

Comment:
@ye-luo I think the leak happens cause these are not deallocated. I could call delete and move on, but in general refactoring towards RAII is the goal to avoid more of these in the future. Hope it helps, thanks!

Reply:
I'm not say one way or another. Having a clear understanding helps us writing better codes. "not deallcoated" can be forgetting to delete in the destructor or the object containing the pointer not gotten freed. So I'd like to understand what is the root cause. Since `delete_iter(h5desc.begin(), h5desc.end());` is supposed to free these pointers.

If you think delete can work, where do you think of to put it. It probably helps me to understand the full problem.

Reply:
Issue #3060 has steps to reproduce and the leak backtrace. You're right, delete could be called anywhere within object scope post usage. The goal of moving towards RAII is not having to worry about manual memory management (and fewer lines of code). Hope it helps.

Reply:
Is this leaky pattern repeated in the main application or does it only occur in the test codes?

Reply:
@prckent that's a good question. For this particular case, I have only see it in the test code. asan failures happen in unit test (real build so far). We can either just try to manage memory manually without modifying internal structures, or refactor towards RAII, so new tests won't have to worry about it. I can always reenable asan to take a look at the failures (mostly coming from `new`). Let me know your thoughts.

```
The following tests FAILED:
	 12 - deterministic-unit_test_estimators (Failed)
	 17 - deterministic-unit_test_einspline (Failed)
	 20 - deterministic-unit_test_wavefunction_trialwf (Failed)
	 21 - deterministic-unit_test_wavefunction_sposet (Failed)
	 22 - deterministic-unit_test_wavefunction_jastrow (Failed)
	 23 - deterministic-unit_test_wavefunction_determinant (Failed)
	 24 - deterministic-unit_test_hamiltonian (Failed)
	 26 - deterministic-unit_test_drivers (Failed)
	 27 - deterministic-unit_test_new_drivers (Failed)
	 29 - deterministic-unit_test_qmcapp (Failed)
```

Reply:
It can be a bit convoluted. This line [`RandomNumberControl::make_seeds();]`(https://github.com/QMCPACK/qmcpack/blob/develop/src/QMCApp/tests/test_rng_control.cpp#L33) is exposed in the the leak, but looking at the API of `RandomNumberControl` I don't know if there is a way to deallocate directly.

```
Direct leak of 2600 byte(s) in 1 object(s) allocated from:
29:     #0 0x4d0e9d in operator new(unsigned long) (/home/wgodoy/workspace/qmcpack/build/tests/bin/test_qmcapp+0x4d0e9d)
29:     #1 0x656c8b in qmcplusplus::RandomNumberControl::make_children() /home/wgodoy/workspace/qmcpack/build/../src/OhmmsApp/RandomNumberControl.cpp:96:24
29:     #2 0x656832 in qmcplusplus::RandomNumberControl::make_seeds() /home/wgodoy/workspace/qmcpack/build/../src/OhmmsApp/RandomNumberControl.cpp:87:3
29:     #3 0x4dbe7a in qmcplusplus::____C_A_T_C_H____T_E_S_T____0() /home/wgodoy/workspace/qmcpack/build/../src/QMCApp/tests/test_rng_control.cpp:33:3
29:     #4 0x54b146 in Catch::TestInvokerAsFunction::invoke() const /home/wgodoy/workspace/qmcpack/build/../external_codes/catch/catch.hpp:11009:9
29:     #5 0x52bee2 in Catch::RunContext::runCurrentTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) /home/wgodoy/workspace/qmcpack/build/../external_codes/catch/catch.hpp:9748:17
29:     #6 0x528952 in Catch::RunContext::runTest(Catch::TestCase const&) /home/wgodoy/workspace/qmcpack/build/../external_codes/catch/catch.hpp:9524:13
29:     #7 0x53d3a9 in Catch::(anonymous namespace)::runTests(std::shared_ptr<Catch::Config> const&) /home/wgodoy/workspace/qmcpack/build/../external_codes/catch/catch.hpp:10073:39
29:     #8 0x53d3a9 in Catch::Session::runInternal() /home/wgodoy/workspace/qmcpack/build/../external_codes/catch/catch.hpp:10266:27
29:     #9 0x53ac1d in Catch::Session::run() /home/wgodoy/workspace/qmcpack/build/../external_codes/catch/catch.hpp:10223:24
29:     #10 0x7fb8d889c0b2 in __libc_start_main /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:308:16
```